### PR TITLE
feat: Add `_experiments.injectBuildInformation` option

### PR DESCRIPTION
**Pull request by mydea**
This allows to gather some build information into `window.SENTRY_BUILD_INFO`.

If enabled (currently opt-in, we may flip this in a future step?), adds the following:

```js
window.SENTRY_BUILD_INFO = {
  deps: [
    "@sentry-internal/eslint-config",
    "@sentry-internal/sentry-bundler-plugin-tsconfig",
    "@sentry/bundler-plugin-core",
    "@swc/jest",
    "@types/jest",
    "@types/webpack4",
    "esbuild",
    "eslint",
    "jest",
    "rollup",
    "ts-node",
    "vite",
    "webpack",
    "webpack4",
  ],
  depsVersions: { vite: 3, webpack: 5 },
  nodeVersion: 14,
};
```

Closes https://github.com/getsentry/sentry-javascript-bundler-plugins/issues/175

### CODE REVIEW

File: packages/bundler-plugin-core/src/utils.ts [hunk 1]
```diff
@@ -7,3 +12,154 @@
...
+  "@angular/core",
+  "vue",
+  "ember-source",
+  "svelte",
+  "@sveltejs/kit",
+  "webpack",
+  "vite",
+  "gatsby",
+  "next",
+  "remix",
```
Comment (AbhiPrasad) [inline packages/bundler-plugin-core/src/utils.ts:103]: should we also do rollup/esbuild?

Comment (mydea) [reply packages/bundler-plugin-core/src/utils.ts:103]: good point! Yes, I'll also add these

File: packages/bundler-plugin-core/src/index.ts [hunk 2]
```diff
@@ -363,9 +367,30 @@ function generateGlobalInjectorCode({
       _global.SENTRY_RELEASES["${key}"]={id:"${release}"};`;
   }
 
+  if (injectBuildInformation) {
+    const buildInfo = getBuildInformation();
+
+    code += `
+      _global.SENTRY_BUILD_INFO=${JSON.stringify(buildInfo)}`;
```
Comment (lforst) [inline packages/bundler-plugin-core/src/index.ts:374]: h: Can we add a semicolon at the end here? Although rare, there are some situations where a missing semicolon will mess up user code.

File: packages/bundler-plugin-core/src/types.ts [hunk 3]
```diff
@@ -198,6 +198,17 @@ export type Options = Omit<IncludeEntry, "paths"> & {
...
+  /**
+   * If set to true, the plugin will inject an additional `SENTRY_BUILD_INFO` variable.
+   * This contains information about the build, e.g. dependencies, node version and other useful data.
+   *
+   * This is still experimental and subject to change!
+   *
+   * Defaults to `false` for now.
+   * @hidden
+   */
+  injectBuildInformation?: boolean;
```
Comment (lforst) [inline packages/bundler-plugin-core/src/types.ts:211]: l: Just a thought. Should we make this an experiment for now? Just so we don't "break" if we decide to unship? Iirc we are still not a 100% sure if we want to go this route.

**Review by lforst — APPROVED** [review]

File: packages/bundler-plugin-core/src/index.ts (hunk 2 above)
Comment (mydea) [reply packages/bundler-plugin-core/src/index.ts:374]: oops, definitely!

File: packages/bundler-plugin-core/src/types.ts (hunk 3 above)
Comment (mydea) [reply packages/bundler-plugin-core/src/types.ts:211]: Yes, probably makes sense! I'll just make it `_experiments.injectBuildInformation` for now 👍

**Comment by mydea** [comment]
Updated this to be confiurable as `_experiments.injectBuildInformation` for now!

**Review by lforst — APPROVED** [review]

**Review by Lms24 — APPROVED** [review]

**Pull request merged**